### PR TITLE
Ember Best Practices: Route Actions

### PR DESCRIPTION
**Pull request by poteto**
[Rendered](https://github.com/DockYard/reefpoints/blob/route-action-helper/source/posts/2016-02-19-best-practices-route-actions.md)

To be published on 19th Feb

### CODE REVIEW

File: source/posts/2016-02-19-best-practices-route-actions.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+tags: ember, javascript, best practices
+summary: "Bubble your closure actions to the route instead of the controller."
+ember_start_version: 2.0.0
+---
+
+In my [previous post](previous-post), I wrote about moving singleton state in Controllers into Services that back components. This means being able to lighten the responsibility of Controllers in your application, and in some cases even remove them completely. However, a common pain point that remains for many is that it isn't really clear what _should_ live on a Controller, if anything. 
+
+![controllers are dead](https://i.imgur.com/TgmUDac.png)
+
+[Locks](locks) has written an excellent [blog post](controllers-are-dead) on this topic. Unless you require query parameters, I'd say that you probably don't need one. That said, I wouldn't go out of your way to creatively avoid using Controllers, and only remove them if it makes sense to do so.
```
Comment (rwjblue) [inline source/posts/2016-02-19-best-practices-route-actions.md]: > I wouldn't go out of your way to creatively avoid using Controllers, and only remove them if it makes sense to do so.

I wouldn't go out of your way to creatively avoid using Controllers, and would only remove them if it makes sense to do so.

File: source/posts/2016-02-19-best-practices-route-actions.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+[Locks](locks) has written an excellent [blog post](controllers-are-dead) on this topic. Unless you require query parameters, I'd say that you probably don't need one. That said, I wouldn't go out of your way to creatively avoid using Controllers, and would only remove them if it makes sense to do so.
+
+## Refactoring away from Controllers
+
+If you do want to remove Controllers from your application given that they are going to be deprecated and removed in the future, great! 
+
+One of the benefits of removing Controllers is that it reduces the amount of "glue code" you need to write to connect your data (route) and its presentation (components). This means a simpler mental model, and reduces the amount of "Ember" you need to write. Instead, you can rely on regular JavaScript you're already familiar with to build your applications.
+
+## Sending closure actions from directly to the route
```
Comment (courajs) [inline source/posts/2016-02-19-best-practices-route-actions.md]: "from directly to the route"?

Comment (poteto) [inline source/posts/2016-02-19-best-practices-route-actions.md]: :microscope: good catch

**Pull request merged**